src/xtopcom/xblockstore/src/xvblockhub.cpp

```diff
@@ -761,6 +762,29 @@ namespace top
                     }
                 }
             }
+            //note:when ask_full_search is true ,here may do heavy job to search all blocks until highest one
+            if(ask_full_search)
+            {
+                for(uint64_t h = m_meta->_highest_genesis_connect_height + 1; h <= m_meta->_highest_commit_block_height; ++h)
```

Comment:
replace ++h with h = m_meta->_highest_genesis_connect_height + 1 ?
or replace for() with while(true) ?